pkg/reconciler/reconciler.go

```diff
@@ -617,6 +617,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.
 		return ctrl.Result{}, err
 	}
 
+	if obj.GetDeletionTimestamp() == nil && !controllerutil.ContainsFinalizer(obj, uninstallFinalizer) {
+		log.V(1).Info("Adding uninstall finalizer")
+		obj.SetFinalizers(append(obj.GetFinalizers(), uninstallFinalizer))
+		if err := r.client.Update(ctx, obj); err != nil {
```

Comment:
Questions:
1. do we still need the `u.Update(updater.EnsureFinalizer(uninstallFinalizer))` at the end of this file?
2. if so, a comment here on why we're not using the updater would be useful.